src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EndpointInfo/ServerEndpointTrackerV2Tests.cs

```diff
@@ -0,0 +1,168 @@
+        {
+            ConnectionMode = DiagnosticPortConnectionMode.Listen
+        });
+
+        private static IServerEndpointStateChecker CreateStateChecker(params ServerEndpointState[] stateCheckResults)
+        {
+            Assert.NotEmpty(stateCheckResults);
+            Mock<IServerEndpointStateChecker> stateChecker = new();
+            var sequence = stateChecker.SetupSequence(s => s.GetEndpointStateAsync(It.IsAny<IEndpointInfo>(), It.IsAny<CancellationToken>()));
+
+            for (int i = 0; i < stateCheckResults.Length; i++)
+            {
+                sequence.ReturnsAsync(stateCheckResults[i]);
+            }
+
+            sequence.ThrowsAsync(new InvalidOperationException("End of test"));
+
+            return stateChecker.Object;
+        }
+
+        private ServerEndpointTrackerV2 CreateTracker(ServerEndpointState stateCheckResult = ServerEndpointState.Active)
+            => new ServerEndpointTrackerV2(CreateStateChecker(stateCheckResult), _timeProvider, _diagPortOptions);
+
+        private ServerEndpointTrackerV2 CreateTracker(IServerEndpointStateChecker stateChecker)
+            => new ServerEndpointTrackerV2(stateChecker, _timeProvider, _diagPortOptions);
+
+
+        [Fact]
+        public async Task Add_RegistersEndpoint()
+        {
+            // Arrange
+            using CancellationTokenSource cts = new CancellationTokenSource(CommonTestTimeouts.GeneralTimeout);
+            using ServerEndpointTrackerV2 tracker = CreateTracker();
+            IEndpointInfo endpoint = Mock.Of<IEndpointInfo>();
+
+            // Act
+            await tracker.AddAsync(endpoint, cts.Token);
+
+            // Assert
+            IEnumerable<IEndpointInfo> activeEndpoints = await tracker.GetEndpointInfoAsync(cts.Token);
+            IEndpointInfo activeEndpoint = Assert.Single(activeEndpoints);
+            Assert.Equal(endpoint, activeEndpoint);
+        }
+
+        [Fact]
+        public async Task PruneEndpointsAsync_Prunes_ErrorEndpoint()
+        {
+            // Arrange
+            using CancellationTokenSource cts = new CancellationTokenSource(CommonTestTimeouts.GeneralTimeout);
+            using ServerEndpointTrackerV2 tracker = CreateTracker(ServerEndpointState.Error);
+
+            IEndpointInfo endpoint = Mock.Of<IEndpointInfo>();
+            await tracker.AddAsync(endpoint, cts.Token);
+
+            EndpointRemovedEventArgs endpointRemovedArgs = null;
+            tracker.EndpointRemoved += (_, args) => endpointRemovedArgs = args;
+
+            // Act
+            await tracker.PruneEndpointsAsync(cts.Token);
+
+            // Assert
+            IEnumerable<IEndpointInfo> activeEndpoints = await tracker.GetEndpointInfoAsync(cts.Token);
+            Assert.Empty(activeEndpoints);
+
+            Assert.NotNull(endpointRemovedArgs);
+            Assert.Equal(ServerEndpointState.Error, endpointRemovedArgs.State);
+            Assert.Equal(endpointRemovedArgs.Endpoint, endpoint);
```

Comment:
I think these parameters should be switched: `endpoint` is the expected value whereas `endpointRemovedArgs.Endpoint` is the tested value.

Reply:
Whoops, good catch